xbmc/settings/dialogs/GUIDialogContentSettings.cpp

```diff
@@ -231,10 +231,11 @@ void CGUIDialogContentSettings::OnSettingAction(const CSetting *setting)
 
       dialog->Open();
       // Selected item has not changes - in case of cancel or the user selecting the same item
-      if (!dialog->IsConfirmed() || dialog->GetSelectedItem() == iSelected)
+      int newSelected = dialog->GetSelectedItem();
+      if (!dialog->IsConfirmed() || newSelected < 0 || newSelected == iSelected)
         return;
 
-      auto selected = labels.at(dialog->GetSelectedItem());
+      auto selected = labels.at(newSelected);
       m_content = static_cast<CONTENT_TYPE>(selected.second);
 
```

Comment:
I dissmissed these warnings in coverity because I believe they're ensured in the API. `IsConfirmed` should only return true when something is selected, which will guarantee the returned value from `GetSelectedItem` is in range.